pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -202,9 +187,73 @@ func (c *Cluster) Start() error {
 	return nil
 }
 
+// reconcile the services while requiring the active mgr to be detected
+func (c *Cluster) ReconcileServices(daemonNameToUpdate string) error {
+	return c.reconcileServices(daemonNameToUpdate, "")
+}
+
+// reconcile the services, if the active mgr is not detected, use the default mgr
+func (c *Cluster) reconcileServices(daemonNameToUpdate, defaultMgr string) error {
+	if daemonNameToUpdate != "" {
+		// If the services are already set to this daemon, no need to attempt to update
+		svc, err := c.context.Clientset.CoreV1().Services(c.clusterInfo.Namespace).Get(context.TODO(), AppName, metav1.GetOptions{})
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				logger.Warningf("mgr services not found, skipping the update")
+				return nil
+			}
+			logger.Errorf("failed to check current mgr service, proceeding to update... err= %v", err)
+		} else {
+			currentDaemon := svc.Spec.Selector[controller.DaemonIDLabel]
+			if currentDaemon == daemonNameToUpdate {
+				logger.Infof("mgr services already set to daemon %q, no need to update", daemonNameToUpdate)
+				return nil
+			}
+			logger.Infof("mgr service currently set to %q, checking if need to update to %q", currentDaemon, daemonNameToUpdate)
+		}
+	}
+
+	mgrMap, err := cephclient.CephMgrMap(c.context, c.clusterInfo)
+	if err != nil {
+		return errors.Wrap(err, "failed to get ceph status for the active mgr")
+	}
+	activeDaemon := mgrMap.ActiveName
+	if activeDaemon == "" {
+		if defaultMgr == "" {
+			return errors.Errorf("active mgr not found")
+		}
+		logger.Infof("active mgr not found, defaulting to %q", defaultMgr)
+		activeDaemon = defaultMgr
+	}
+	if daemonNameToUpdate != "" && daemonNameToUpdate != activeDaemon {
+		logger.Infof("skipping mgr update since the active mgr is %q, but cannot be updated by local mgr %q", activeDaemon, daemonNameToUpdate)
+		return nil
+	}
+
+	logger.Infof("setting active mgr: %q", activeDaemon)
```

Comment:
I think it's confusing, this gives the impression we are changing the active manager in Ceph where we just configure service with it.